snapcraft/how-to/qt-applications/qt-applications.md

```diff
@@ -1,265 +1,486 @@
-* **Upgrades are safe**
-  If your app fails to upgrade, users automatically roll back to the previous revision.
+For a brief overview of the snap creation process, including how to install Snapcraft and how it's used, see [Snapcraft overview](/t/snapcraft-overview/8940). For a more comprehensive breakdown of the steps involved, take a look at [Creating a snap](/t/creating-a-snap/6799).
+[/note]
 
-### Build a snap in 20 minutes
+## Creating a new Qt 5 and KDE Frameworks 5 based snap
 
-Typically this guide will take around 20 minutes and will result in a working Qt5 application in a snap. Once complete, you'll understand how to package Qt5 applications as snaps and deliver them to millions of Linux users. After making the snap available in the store, you'll get access to installation metrics and tools to directly manage the delivery of updates to Linux users.
+Snaps are defined in a single YAML file named *snapcraft.yaml* located in the root folder of your project. This YAML file describes the application, its dependencies and how it should be built.
 
-> ⓘ For a brief overview of the snap creation process, including how to install *snapcraft* and how it's used, see [Snapcraft overview](/t/snapcraft-overview/8940). For a more comprehensive breakdown of the steps involved, take a look at [Creating a snap](/t/creating-a-snap/6799).
+For this how-to, we will create and build a snap of KDE's calculator application [KCalc](https://apps.kde.org/kcalc/). We will also explore how we can submit our snap to the Snap Store.
 
-## Getting started
-
-Snaps are defined in a single YAML file placed in the root folder of your project. The following example shows the entire *snapcraft.yaml* file for KCalc. Don't worry, we’ll break this down.
-
-### KCalc Snap
-
-Snaps are defined in a single yaml file placed in the root of your project. The Kcalc example shows the entire snapcraft.yaml for an existing project. We'll break this down.
-
-[details=snapcraft.yaml for KCalc]
+We're going to start by creating a *snapcraft.yaml* file that contains the minimum needed to build a working version of KCalc. We'll explain the contents of this file line-by-line in the following sections of this guide.
 
+[details=Our minimal snapcraft.yaml for KCalc]
 ```yaml
-name: kcalc
-version: '19.08.0'
-grade: stable
-adopt-info: kcalc
+name: kcalc-example
+adopt-info: build-kcalc
+grade: devel
 
-confinement: strict
-base: core18
+base: core22
+confinement: devmode
 
 apps:
-  kcalc:
+  kcalc-example:
     common-id: org.kde.kcalc.desktop
-    command: kcalc
+    command: usr/bin/kcalc
     extensions:
       - kde-neon
-    plugs:
-      - kde-frameworks-5-plug
-      - home
-      - opengl
-      - network
-      - network-bind
-      - pulseaudio
-
-slots:
-  session-dbus-interface:
-    interface: dbus
-    name: org.kde.kcalc.desktop
-    bus: session
 
 parts:
-  kcalc:
+  build-kcalc:
+    source: https://invent.kde.org/utilities/kcalc/-/archive/v23.08.5/kcalc-v23.08.5.tar.gz
```

Comment:
This how-to uses KCalc v23.08.5 - it's the latest (if not final) version that supports Qt 5, as v24 onwards seems to be purely Qt 6 based. I wasn't sure whether it was worth mentioning it here.

Reply:
Thank you so much for this.  I think it may be worth mentioning, but it's not vital.